SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs

```diff
@@ -0,0 +1,105 @@
+use std::{io::Write, path::PathBuf};
+use coswid_core::*;
+use digest::Digest;
+use clap::Parser;
+use anyhow::{Result, anyhow};
+
+const STM_BIN_GUID: &str = "C7600D63-09A2-4A30-A105-DAA22DE2D0FE";
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+/// Generates measurement hashes for the STM firmware binary file and wraps it in a swid+cbor RIM
+/// containing the following hashes: SHA256, SHA384, SHA512, SM3-256.
+/// 
+pub struct Args {
+    /// Path to the STM firmware binary file to be measured
+    
+    pub file: PathBuf,
+    /// Version of the STM firmware binary file
+    #[arg(short, long)]
+    pub rim_version: Option<String>,
+    /// Path to place the generated RIM file. Defaults to the current directory.
+    #[arg(short, long)]
+    pub output: Option<PathBuf>,
+    #[arg(long)]
+    /// Company name for registering the Software Creator in the RIM
+    pub company_name: String,
+    /// Company URL for registering the Software Creator in the RIM
+    #[arg(long)]
+    pub company_url: String,
+}
+
+fn main() -> Result<()> {
+    let args = Args::parse();
+
+    let measurement_payloads = generate_measurements(&args.file)?;
+
+    let cosesign1_payload: ConciseSwidTag<(), Vec<FileMeasurement>> = 
+        ConciseSwidTag::new(STM_BIN_GUID, 0, format!("{} STM Binary", args.company_url))
+            .with_software_version(args.rim_version.unwrap_or("1.0.0".to_string()))
+            .with_version_scheme("1")
+            .with_software_meta(SoftwareMetaEntry::new().with_product("STM Binary"))
+            .with_entity(EntityEntry::new(args.company_name).with_reg_id(args.company_url).with_role("tag-creator").with_reg_id("software-creator"))
+            .with_payload(measurement_payloads);
+
+    let cosesign1: CoseSign1<(), ConciseSwidTag<(), Vec<FileMeasurement>>> = CoseSign1::new(
+        ProtectedHeader::new(-39, "application/swid+cbor"),
+        UnprotectedHeader::new(),
+        cosesign1_payload,
+        "F".repeat(256),
+    );
+
+    let mut buffer = Vec::new();
+    minicbor::encode(&cosesign1, &mut buffer).unwrap();
+    let mut file = std::fs::File::create(args.output.unwrap_or(PathBuf::from("RIM.bin"))).unwrap();
+    file.write_all(&buffer).unwrap();
+    Ok(())
+}
+
+
+
+pub fn generate_measurements(file: &PathBuf) -> Result<Vec<FileMeasurement>> {
```

Comment:
Does this need to be public?

Reply:
Nope. everything in main does not. I've removed.